src/sentry/sentry_metrics/indexer/indexer_consumer.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from typing import Any, Dict
+
+from django.conf import settings
+
+from sentry.sentry_metrics import indexer
+from sentry.sentry_metrics.indexer.base import UseCase
+from sentry.utils import json, kafka_config
+from sentry.utils.batching_kafka_consumer import AbstractBatchWorker
+from sentry.utils.kafka import create_batching_kafka_consumer
+from sentry.utils.pubsub import KafkaPublisher
+
+logger = logging.getLogger(__name__)
+
+snuba_metrics = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS]
+
+
+def get_metrics_consumer(topic=None, **options) -> None:
+    return create_batching_kafka_consumer(
+        {"ingest-metrics"}, worker=MetricsIndexerWorker(), **options
+    )
+
+
+class MetricsIndexerWorker(AbstractBatchWorker):
+    def process_message(self, message):
+        original: Dict[str, Any] = json.loads(message.value())
+        new_message = original.copy()
+        project_id = int(new_message["project_id"])
+
+        metric_id = indexer.resolve(project_id, UseCase.METRIC, new_message["name"])
+
+        tags = new_message["tags"]
+        new_tags = {}
+        for tag_k, tag_v in tags.items():
+            new_k = indexer.resolve(project_id, UseCase.TAG_KEY, tag_k)
+            new_v = indexer.resolve(project_id, UseCase.TAG_VALUE, tag_v)
+            new_tags[new_k] = new_v
+
+        new_message["tags"] = new_tags
+        new_message["metric_id"] = metric_id
+        new_message["retention_days"] = 90
+        return new_message
+
+    def flush_batch(self, batch):
```

Comment:
I wonder if we can reduce the number of calls to the indexer by doing the transformation per-batch rather than per-message (collecting all strings in the batch and filtering out duplicates).
On the other hand, if the actual string indexer has an in-memory cache, that may not be necessary.